File: src/main/java/wraith/fabricaeexnihilo/api/registry/IMeshRegistry.java
package wraith.fabricaeexnihilo.api.registry;

import net.minecraft.util.Identifier;
import wraith.fabricaeexnihilo.modules.sieves.MeshProperties;
import wraith.fabricaeexnihilo.util.Color;

import java.util.List;

public interface IMeshRegistry extends IRegistry<MeshProperties> {

    default boolean register(Identifier identifier, int enchantability, String displayName, Color color, Identifier ingredient) {
        return register(new MeshProperties(identifier, enchantability, displayName, color, ingredient));
    }

    void registerItems();

    List<MeshProperties> getAll();

}

File: src/main/kotlin/exnihilofabrico/api/registry/IMeshRegistry.kt
package exnihilofabrico.api.registry

import exnihilofabrico.modules.sieves.MeshProperties
import exnihilofabrico.util.Color
import net.minecraft.item.Item
import net.minecraft.util.Identifier
import net.minecraft.util.registry.Registry

interface IMeshRegistry: IRegistry<MeshProperties> {
    fun register(identifier: Identifier, enchantability: Int, displayName: String, color: Color, ingredient: Identifier) =
        register(MeshProperties(identifier, enchantability, displayName, color, ingredient))

    fun registerItems(itemRegistry: Registry<Item>)

    fun getAll(): List<MeshProperties>

}